Добавил подробный просмотр информации о фильме:

novopashinwm opened the pull request:
1) Пришлось добавить полей в класс Movie.kt, чтобы соответствовало проекту.
2) Понимаю, что сделал аляписто в movie_details_fragment.xml с LinearLayout, мне нужна более подрбная информация как более аккуратно выводить данные типа таблица.
3) Удалось скруглить кнопку для просмотра фильма, но не удалось скруглить фотки артистов. Наверняка какое-то свойство есть.
4) Как в MockDetailsRepository.kt вызвать getString, чтобы получить доступ к ресурсам - раньше я получал через Resources.getString - сейчас этот метод depricated.
5) Для просмотра артистов пришлось создать свои item в классах и верстке, а также контейнер для хранения этих артистов.

novopashinwm committed:
Добавил подробный просмотр информации о фильме:
1) Пришлось добавить полей в класс Movie.kt, чтобы соответствовало проекту.
2) Понимаю, что сделал аляписто в movie_details_fragment.xml с LinearLayout, мне нужна более подрбная информация как более аккуратно выводить данные типа таблица.
3) Удалось скруглить кнопку для просмотра фильма, но не удалось скруглить фотки артистов. Наверняка какое-то свойство есть.
4) Как в MockDetailsRepository.kt вызвать getString, чтобы получить доступ к ресурсам - раньше я получал через Resources.getString - сейчас этот метод depricated.
5) Для просмотра артистов пришлось создать свои item в классах и верстке, а также контейнер для хранения этих артистов.

AndroidStudentClub commented:
Сразу отвечу на вопросы:
3) Для скругления артистов можно применить стиль 
  &lt;style name="ShapeAppearanceOverlay.App.CornerSize50Percent" parent=""&gt;        &lt;item name="cornerSize"&gt;50%&lt;/item&gt;    &lt;/style&gt;
4) Для получения строковых ресурсов нужен будет context. НО! Использовать context внутри репозитория - это плохая практика. Так что 1) Для мока достаточно просто написать хардкодом строки (после интеграции мы их удалим) 2) Отвечая на вопрос как обычно получить строковый ресурс можно использовать https://developer.android.com/reference/android/content/res/Resources#getString(int) - вроде он не депрекейтед. Ну или можно использовать ContextCompat

AndroidStudentClub commented on app/src/main/java/ru/androidschool/intensiv/ui/movie_details/ArtistItem.kt:
```diff
@@ -0,0 +1,23 @@
+
+    override fun bind(viewHolder: GroupieViewHolder, position: Int) {
+        viewHolder.FIO.text = content.fullname
+        Picasso.get()
```
Вот этот код повторяется - можно сделать проще, написав один раз extension для ImageView внутри которого будет эта логика. В итоге будет что-то типа imagePreview.loadImage(url)

novopashinwm replied:
Добавил extention.

AndroidStudentClub commented on app/src/main/java/ru/androidschool/intensiv/ui/movie_details/MovieDetailsFragment.kt:
```diff
@@ -1,6 +1,47 @@
+
+        val movie = MockDetailsRepository.getDetailMovie()
+
+        Picasso.get().load(web_path+movie.poster).into(header_image_detail)
```
Вот этот код повторяется - можно сделать проще, написав один раз extension для ImageView внутри которого будет эта логика. В итоге будет что-то типа imagePreview.loadImage(url)

novopashinwm replied:
Добавил extension

AndroidStudentClub commented on app/src/main/res/layout/item_actor.xml:
```diff
@@ -0,0 +1,25 @@
+        android:layout_height="@dimen/image_preview_side"
+        android:layout_margin="@dimen/material_margin_small"
+        android:scaleType="centerCrop"
+        app:shapeAppearanceOverlay="@style/ShapeAppearanceOverlay.App.CornerSize10Percent"
```
Для скругления надо написаь стиль похожий на 
ShapeAppearanceOverlay.App.CornerSize10Percent - только внутри поменять 10 на 50%

novopashinwm replied:
Создал свой стиль и применил его к этому image

AndroidStudentClub commented on app/src/main/res/layout/item_actor.xml:
```diff
@@ -0,0 +1,25 @@
+        tools:src="@tools:sample/avatars" />
+
+    <TextView
+        android:id="@+id/FIO"
```
Обычно капсом не называют, лучше что-то типа name например

novopashinwm replied:
Вместо FIO сделал fullname.

AndroidStudentClub commented on app/src/main/res/layout/movie_details_fragment.xml:
```diff
@@ -1,12 +1,165 @@
+                tools:text="Начало"
+                android:text="@string/mock_details_description"
+                tools:textColor="@color/grey" />
+            <ImageButton
```
На мой взгляд для лайка лучше использовать CheckBox. У него автоматически будет из коробки работа со состояниями (checked/unchecked). И создать кастомный селектор в виде пустого и закрашенного сердечка
https://stackoverflow.com/a/3193127/1462969

novopashinwm replied:
Переделал на CheckBox.

AndroidStudentClub commented on app/src/main/res/layout/movie_details_fragment.xml:
```diff
@@ -1,12 +1,165 @@
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content" />
+        </LinearLayout>
+        <LinearLayout
```
Есть несколько способов сделать такую вертску типа
Название-------Значение
Название-------Значение
1) Можно было LinearLayout а внутри несколько RelativeLayout - будет похоже на текущее решение
2) Использовать GridLayout https://developer.android.com/reference/androidx/gridlayout/widget/GridLayout - он позовляет расположить в виде таблицы элементы (наприер указаать сколько столбцов и строк) - но на самом деле это устаревший подход - но в целом тоже рабочий.

AndroidStudentClub replied:
Но такой подход тоже ок, можно было бы еще сделать
3)RecyclerView - создать одну ячейку и потом 3 раза добавить данные
4)Сделать композитную view которая внутри будет иметь 2 TextView
И положить эти 3 композитных view в LinearLayout -тогда будет меньше кода в верстке

AndroidStudentClub replied:
Вот тут пример https://code.tutsplus.com/tutorials/creating-compound-views-on-android--cms-22889
Еще можно посмотреть как сделан SearchBar - в проекте - это тоже пример композитной view. То есть это просто обертка над кусочком layout

novopashinwm replied:
Выбрал LinearLayout и несколько RelativeLayoute внутри него.

AndroidStudentClub commented on app/src/main/res/layout/movie_details_fragment.xml:
```diff
@@ -1,12 +1,165 @@
+            android:orientation="horizontal">
+
+            <androidx.recyclerview.widget.RecyclerView
+                android:id="@+id/artists_recycler_view"
```
Да, для артистов нормально сделать RecyclerView

AndroidStudentClub commented on app/src/main/res/layout/movie_details_fragment.xml:
```diff
@@ -1,12 +1,165 @@
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="100dp"
```
Размеры, цвета, строки рекомендуется выносить в ресурсы

novopashinwm replied:
Вынес.

AndroidStudentClub commented on app/src/main/java/ru/androidschool/intensiv/ui/movie_details/MovieDetailsFragment.kt:
```diff
@@ -1,6 +1,47 @@
+        year_value.text = movie.year
+
+        val newArtistList = listOf(
+            ArtistCardContainer(
```
Не совсем понял зачем тут контейнер используется. Можно было бы и без него обойтись, по сути для списка актеров нужно сделать все то же самое что и с сериалами, только разница  в том, что RecyclerView горизонтальный.

novopashinwm replied:
Да я скопипастил логику с MovieCardItem. Это имело бы смысл, если бы всякий раз в этот контейнер я бы передавал разные названия и таких контейнеров было бы несколько. А так да согласен - мне два recyclerview для артистов не надо - сделал один.

AndroidStudentClub commented on app/src/main/res/layout/item_card_actors.xml:
```diff
@@ -0,0 +1,23 @@
+    tools:background="@color/blueDark">
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/actor_items_container"
```
Не совсем понял зачем тут контейнер используется. Можно было бы и без него обойтись, по сути для списка актеров нужно сделать все то же самое что и с сериалами, только разница в том, что RecyclerView горизонтальный.

novopashinwm replied:
Убрал контейнер - он тут не нужен.

AndroidStudentClub reviewed (approved):
Проверил, не совсем понял зачем опять 2 RecyclerView для 1-го списка актеров

novopashinwm committed:
Исправил ошибки:
1) Вынес в extentions загрузку картинок.
2) Убрал ArtistCardContainer.kt и работу с 2 RecyclerView.
3) Заменил работу с like на контрол CheckBox
4) Изменил табличку на RelativeLayout.

novopashinwm committed:
1) Вынес в ресурсы высоту RecyclerView с артистами.
